### ISSUE TITLE
 *** Break *** segmentation violation

### ISSUE THREAD

**Issue by Hachiman512** [fm140905/coincidence#1]
I am getting this error with my .bin file. I have defined all the parameters and settings according to my CAEN module in that CsCalinbration.json file. I have tried your test data "channel0.bin" which is working perfectly fine but I don't know what's wrong with my data file. Please help me in this regard:)

This is the error output I am getting::

Open file: DataR_run_9.bin
 Reading data: 

 *** Break *** segmentation violation

===========================================================
There was a crash.
This is the entire stack trace of all threads:
===========================================================
#0  0x00007f7c346fcc37 in wait4 () from /lib64/libc.so.6
#1  0x00007f7c346711fb in do_system () from /lib64/libc.so.6
#2  0x00007f7c34f07cf9 in TUnixSystem::StackTrace() () from /home/anil/software/root/install/lib/libCore.so
#3  0x00007f7c34f076b4 in TUnixSystem::DispatchSignals(ESignals) () from /home/anil/software/root/install/lib/libCore.so
#4  <signal handler called>
#5  0x000000000041c41d in Event::getPulseHeight(std::vector<double, std::allocator<double> > const&, int&) ()
#6  0x000000000041cb1e in Event::parse(char const*, unsigned long long&, std::vector<double, std::allocator<double> >&) ()
#7  0x000000000041a258 in Channel::loadEvents() ()
#8  0x00000000004057ac in main ()
===========================================================

The lines below might hint at the cause of the crash. If you see question
marks as part of the stack trace, try to recompile with debugging information
enabled and export CLING_DEBUG=1 environment variable before running.
You may get help by asking at the ROOT forum https://root.cern/forum
Only if you are really convinced it is a bug in ROOT then please submit a
report at https://root.cern/bugs Please post the ENTIRE stack trace
from above as an attachment in addition to anything else
that might help us fixing this issue.
===========================================================
#5  0x000000000041c41d in Event::getPulseHeight(std::vector<double, std::allocator<double> > const&, int&) ()
#6  0x000000000041cb1e in Event::parse(char const*, unsigned long long&, std::vector<double, std::allocator<double> >&) ()
#7  0x000000000041a258 in Channel::loadEvents() ()
#8  0x00000000004057ac in main ()
===========================================================

**Comment by fm140905** [comment]
Hi @Hachiman512 it seems the program failed to parse the code. what version of CoMPASS did you use to acquire the binary data?

**Comment by anilsharmanp** [comment]
I am using CoMPASS version 1.

By the way, I figured out the issue. Actually, the problem was in my .bin data file. The problem is that I stored my data in list mode instead the waveform mode. Thanks to your CoMPASSBIN2CSV program, I was able to read out your test data and my own data format. Then I understood that I needed to store my data in waveform not in list mode.

Thanks a lot for your response, I appreciate that:)
